lightning/src/ln/features.rs

```diff
@@ -318,15 +318,13 @@ mod sealed {
 
 			impl <T: $feature> Features<T> {
 				/// Set this feature as optional.
-				pub fn $optional_setter(mut self) -> Self {
+				pub fn $optional_setter(&mut self) {
 					<T as $feature>::set_optional_bit(&mut self.flags);
-					self
 				}
 
 				/// Set this feature as required.
-				pub fn $required_setter(mut self) -> Self {
+				pub fn $required_setter(&mut self) {
 					<T as $feature>::set_required_bit(&mut self.flags);
-					self
 				}
```

Comment:
Could we still support the chained-notation by taking a `&mut self` and returning `&Self`?

Reply:
Oh, right, I'd tried that and realized you still have to declare the variable but you're right, you can at least then be able to chain two setters which is nice, will do.

Reply:
In addition to the updated mutators, I wonder if we should have a different notation for initialization.  Something like:

```rust
let features = InvoiceFeatures:from_bits(VariableLengthOnion::required() | PaymentSecret::required());
```

Would still need to do it in such away that it could be checked at compile time, of course.

Reply:
Yea, that'd be cool, I'd think we'd have to use macros, tho maybe we can override `|`, as ugly as that is.